pkg/karmadactl/apply.go

```diff
@@ -0,0 +1,190 @@
+
+		Alpha Disclaimer: the --prune functionality is not yet complete. Do not use unless you are aware of what the current state is. See https://issues.k8s.io/34274.`))
+
+	applyExample = templates.Examples(i18n.T(`
+		# Apply the configuration in pod.json to a pod
+		%[1]s apply -f ./pod.json
+
+		# Apply resources from a directory containing kustomization.yaml - e.g. dir/kustomization.yaml
+		%[1]s apply -k dir/
+
+		# Apply the JSON passed into stdin to a pod
+		cat pod.json | %[1]s apply -f -
+
+		# Note: --prune is still in Alpha
+		# Apply the configuration in manifest.yaml that matches label app=nginx and delete all other resources that are not in the file and match label app=nginx
+		%[1]s apply --prune -f manifest.yaml -l app=nginx
+
+		# Apply the configuration in manifest.yaml and delete all the other config maps that are not in the file
+		%[1]s apply --prune -f manifest.yaml --all --prune-whitelist=core/v1/ConfigMap`))
+)
+
+// NewCmdApply creates the `apply` command
+// TODO(@carlory): use kcmdutil.factory as an input argument instead of the karmada config,
+// in order to make it act like the kubectl apply command.
+func NewCmdApply(karmadaConfig KarmadaConfig, parentCommand string) *cobra.Command {
+	o := &CommandApplyOptions{}
+	ioStreams := genericclioptions.IOStreams{In: getIn, Out: getOut, ErrOut: getErr}
+	flags := apply.NewApplyFlags(nil, ioStreams)
+	kubeConfigFlags := NewConfigFlags(true).WithDeprecatedPasswordFlag()
+	cmd := &cobra.Command{
+		Use:                   "apply (-f FILENAME | -k DIRECTORY)",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Apply a configuration to a resource by file name or stdin"),
+		Long:                  applyLong,
+		Example:               fmt.Sprintf(applyExample, parentCommand),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			// factory must be initialized before the command runs
+			restConfig, err := karmadaConfig.GetRestConfig(o.KarmadaContext, o.KubeConfig)
+			if err != nil {
+				return err
+			}
+			kubeConfigFlags.WrapConfigFn = func(config *restclient.Config) *restclient.Config { return restConfig }
+			flags.Factory = kcmdutil.NewFactory(kubeConfigFlags)
+			return nil
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			// coverts flags to the kubectl apply options
+			o, err := flags.ToOptions(cmd, parentCommand, args)
+			kcmdutil.CheckErr(err)
+			// generate and inject policies
+			o.PreProcessorFn = func() error {
+				return generateAndInjectPolices(flags.Factory, o)
+			}
+			kcmdutil.CheckErr(o.Validate(cmd, args))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+	flags.AddFlags(cmd)
+	cmd.Flags().StringVar(&o.KubeConfig, "kubeconfig", "", "Path to the control plane kubeconfig file.")
+	cmd.Flags().StringVar(&o.KarmadaContext, "karmada-context", "", "Name of the cluster context in control plane kubeconfig file.")
+	// Fixme(carlory): kubectl uses kubeConfigFlags as global command-line options (applies to all commands).
```

Comment:
The actions of `karmadactl` are different form `kubectl`, so it seems not to apply to all subcmd of `karmadactl`.